io.github.ec_.Quake3e.OpenArena.yaml

```diff
@@ -0,0 +1,73 @@
+app-id: io.github.ec_.Quake3e.OpenArena
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+# CHANGED: Let it look up the command via $PATH
+command: launch
+modules:
+  # CHANGED: Module name isn't displayed to users and convention says lowercase
+  - name: openarena
+    buildsystem: simple
+    sources:
+      # CHANGED: Pull direct from the upstream source
+      - type: git
+        url: "https://github.com/ec-/Quake3e"
+        commit: "f69865508ef4f3d63bcc82c8505d661a1c290f92"
+        tag: "2022-10-14"
+        x-checker-data: # Allow bot to automatically make PRs for new releases
+          type: git
+          tag-pattern: "^(\\d{4}-\\d{2}-\\d{2})$"
+    # CHANGED: Removed config-opts because it's for ./configure, not bare make
+    build-commands:
+      - make USE_CODEC_VORBIS=1 USE_INTERNAL_OGG=1 USE_VOIP=1 USE_CODEC_OPUS=1 USE_OPENAL=1 USE_MUMBLE=1 USE_FREETYPE=1 USE_INTERNAL_LIBS=1 BUILD_SERVER=0COPYDIR=/app/share/openarena COPYBINDIR=/app/bin DEFAULT_BASEDIR=/app/share/openarena USE_CURL=1 USE_CURL_DLOPEN=0
+      - make install "DESTDIR=${FLATPAK_DEST}/bin"
+      # CHANGED: Use ${FLATPAK_DEST}/bin because Quake3e is one of those weird
+      #          exceptions that doesn't follow platform conventions.
+  - name: openarena-data
+    buildsystem: simple
+    sources:
+      - type: archive
+        url: "https://downloads.sourceforge.net/project/oarena/openarena-0.8.8.zip"
+        sha256: "5a8faf7f5b51f351b0a1618c06b6b98a5f1a6758f1d39818de2c87df2a0bac4a"
+    build-commands:
+      - mv baseoa missionpack /app/bin/
+  - name: launch-script
```

Comment:
The `launch` script uses `quake3e.x64`. You need to change it so it also works with aarch64.

Reply:
...or blacklist aarch64 if the Quake3e codebase can't produce a working build for it.

Reply:
> ...or blacklist aarch64 if the Quake3e codebase can't produce a working build for it.

I'll try that, since I don't have any RPi4 to test if it will work

Reply:
You could see if it'll run under emulation.

Reply:
> The `launch` script uses `quake3e.x64`. You need to change it so it also works with aarch64.

I dondon't think Quake3e will run on Arm, is there a way to specify it is x86_64 only?

Reply:
> You could see if it'll run under emulation.

Not really, my PC is too old for that, it struggles with Gnome already. Anyways, the FlatPak builder documentation doesn't specify a lot about architectures, am I missing something?

Reply:
You specify it at a different layer than `flatpak-builder` using a file named `flathub.json`.

See https://github.com/flathub/flathub/wiki/App-Maintenance#flathubjson

Reply:
> You could see if it'll run under emulation.

Not really, my PC is too old for that, it struggles with Gnome already. Anyways, the FlatPak builder documentation doesn't specify a lot about architectures, am I missing something?Ah, thanks, I'll try. I'll come back tomorrow if I could make it work.

Reply:
> You specify it at a different layer than `flatpak-builder` using a file named `flathub.json`.
> 
> See https://github.com/flathub/flathub/wiki/App-Maintenance#flathubjson

Ah, thanks, I'll try. I'll come back tomorrow if I could make it work.

Reply:
bot, build io.github.ec_.Quake3e.OpenArena

Reply:
I have hard time believing it won't build for aarch64 given 1) it builds, as proven by all builds so far 2) original OpenArena is also available for aarch64).

Reply:
Yeah but I don't know how, so I'd like to Keep It Simple, as long as it is not a requirement, also, I corrected everything you pointed out, now the .desktop, icon, and appdata.xml files are no longer in a ZIP, I changed the name and ID, and the bot no longer builds for aarch64, so I'm excited because this might be the last comment I make before my first port gets into Flathub officially!

Reply:
It's a requirement in my opinion. There's no reason to disable aarch64 if it's known it works. It's not difficult to set a variable with a case statement in bash based on `uname -m` output.

Reply:
> It's a requirement in my opinion. There's no reason to disable aarch64 if it's known it works. It's not difficult to set a variable with a case statement in bash based on `uname -m` output.

Done, aarch64 & PPC64 (I can't test if they work) can now be executed by the script, lemme just test if it runs normally, because I need to see if it passes the CVARS correctly.

Reply:
there is no ppc64 support anyway on flathub so it's actually irrelevant.

Reply:
but then the package is still only build on x86_64 due to the flathub.json content.

Reply:
> there is no ppc64 support anyway on flathub so it's actually irrelevant.

I'm future proof?

Reply:
> but then the package is still only build on x86_64 due to the flathub.json content.

I already made it .bak so the bot won't use it

Reply:
> It's a requirement in my opinion. There's no reason to disable aarch64 if it's known it works. It's not difficult to set a variable with a case statement in bash based on `uname -m` output.

Sorry, maybe its a bumb question, but, the quake3e binary ends in .x64, which is not standard, it should be.x86_64, how do I know what the aarch64 build would be? I tried checking the flathub bot logs but that didn't help much